research/delf/delf/python/datasets/sfm120k/sfm120k.py

```diff
@@ -0,0 +1,224 @@
+# Copyright 2021 The TensorFlow Authors All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Structure-from-Motion dataset (Sfm120k)."""
+
+import os
+import pickle
+import tensorflow as tf
+
+from delf.python.datasets import tuples_dataset
+from delf.python.datasets import utils
+
+
+def id2filename(id, prefix):
+  """Creates a training image path out of its CID name.
+
+  Used for the image mapping in the Sfm120k datset.
+
+  Args:
+    id: String, image id.
+    prefix: String, root directory where images are saved.
+
+  Returns:
+   filename: String, full image filename.
+  """
+  if prefix:
+    return os.path.join(prefix, id[-2:], id[-4:-2], id[-6:-4], id)
+  else:
+    return os.path.join(id[-2:], id[-4:-2], id[-6:-4], id)
+
+
+class _Sfm120k(tuples_dataset.TuplesDataset):
+  """Structure-from-Motion (Sfm120k) dataset instance.
+
+  The dataset contains the image names lists for training and validation,
+  the cluster ID (3D model ID) for each image and indices forming
+  query-positive pairs of images. The images are loaded per epoch and resized
+  on the fly to the desired dimensionality. Extends
```

Comment:
"Extends tuples_dataset.TuplesDataset."
This can be removed, it's already obvious